src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.cs

```diff
@@ -252,7 +254,7 @@ private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
                         return;
                     }
 
-                    context.ReportDiagnostic(awaitOperation.Operation.CreateDiagnostic(rule, ruleMessageMethod, ruleMessagePreferredMethod));
+                    context.ReportDiagnostic(invocation.CreateDiagnostic(rule, ruleMessageMethod, ruleMessagePreferredMethod));
```

Comment:
I noticed the fixer was catching the whole await operation. Now the fixer will only receive the actual method invocation: `stream.ReadAsync(...)` or `stream.WriteAsync(...)`.